### ISSUE TITLE
Calendar API for parish page schedule updates

### ISSUE THREAD

**Issue by lucastbelem** [themeetinghouse/web#1002]
Parishes are meeting at very different times these days. We need something that allows us to show their schedule better. Garnet had a great idea. In the area highlighted in the screenshot below, we'd like to have data that updates automatically each week to display that parish's next meeting. For example, it should say "Sunday May 16 at 10am.". 

The second part of this is where the data is coming from. Would love your ideas for this. Garnet thought of some kind of calendaring app or even google calendar's API. We could create individual google calendars for each parish (they may already exist), pastors can manage it themselves, and then the data is sent to the website. If something gets cancelled, the website knows and updates it accordingly. 

Thoughts?

**Comment by itbel** [comment]
I suggest we create some new types and add the ability to change them via the admin screens. @GeorgeBellTMH 

```
type TMHLocation { 
    address: String
    latitude: Number
    longitude: Number
    postal_code: String
    city: String
    province: String
}


type TMHParish { 
    name: String
    parishID: String
    location: TMHLocation
    pastor_email: AWSEmail # or contact_pastor: TMHPerson
    pastors: [TMHPerson]
    staff: [TMHPerson]
    events: [FBEvent] # Maybe? I'm not sure about the implementation for this.
    meeting_date: TMHMeetingDate
    is_hidden: Boolean
    is_online: Boolean
    is_active: Boolean
}


enum Occurrence {
    # What kind of flexibility do we want here?
    WEEKLY
    BIWEEKLY
    MONTHLY
}

type TMHMeetingDate { 
    start_date: AWSTimeStamp
    end_date: AWSTimeStamp
    # We can either use occurrence, or ask for 2 dates and calculate based on that? 
    occurrence: Occurrence
    # next_date: AWSTimeStamp
}



type TMHPhone  {
    number: AWSPhone
    extension: Number
}

type TMHPerson {
    first_name: String
    last_name: String
    position: String
    email: AWSEmail
    phone: TMHPhone

    # maybe make this enum?
    is_coordinator: Boolean
    is_staff: Boolean
    is_counsellor: Boolean
    is_overseer: Boolean
    is_youth: Boolean

    image: AWSURL
}

**Comment by lucastbelem** [comment]
@itbel I wonder if this fits in with our admin interface improvements project. One of the things I'd like to be able to do is create/set parishes from admin and determine all details related to it there too (i.e. date, time, location, etc).

**Comment by itbel** [comment]
Yes this would fall under that. It would move some JSON data into admin interface.

**Comment by lucastbelem** [comment]
Interim solution:

- Make parish meeting frequencies configurable allowing admins to set:
- [ ] How often a parish meets (i.e. interval of days)
- [ ] On which date of the week (i.e. Sundays)
- [ ] What time a parish meets (i.e. 10am)

Display the text on black box on parish pages as follows:

Example 1
- Line 1: Sunday May 29 @ 10:00am
- Line 2: Every other week

Example 2
- Line 1: Sunday May 29 @ 3:00pm
- Line 2: Weekly

Example 3
- Line 1: Sunday May 29 @ 2:00pm
- Line 2: 1st and 3rd Sundays of each month

Frequencies to account for:
- Weekly
- Every other week (bi-weekly)
- Monthly
- 1st and 3rd Sunday (this sounds like bi-weekly. but the frequency in days actually changes depending on how may weeks a month has)
- Irregular (I don't know if there is any logic you can account for here, but this is when the parish has no regular rhythm)

Once you get this data, you can use it to make the "add to calendar" for each parish work.

**Comment by itbel** [comment]
What about the holiday case?

**Comment by lucastbelem** [comment]
I don't think we have a solid way to account for that because I don't if when there is a holiday they cancel or reschedule. So I can just use the irregular rhythm for that.

**Comment by GeorgeBellTMH** [comment]
Optionally...we could just use Facebook...

**Comment by GeorgeBellTMH** [comment]
We already have the event pulling feature...so we would just need a way to identify Sunday services in a way that we can filter them from events...maybe any event named "Weekly Teaching" or "Sunday Service"...

**Comment by GeorgeBellTMH** [comment]
I think we can just set them all up from the business interface...really quick...nice interface...can be farmed out to whoever...bonus is that Facebook now has all our meeting times...

**Comment by lucastbelem** [comment]
My other concern with facebook is that we've had some significant issues pulling event data from them. Some hiccups with missing pieces and, once or twice, events missing altogether. If we use facebook, I'm concerned that we'd increase the volume of things that potentially break. Maybe we can use facebook instead of the longer term calendar or TMH admin interface, once we figure facebook API so it works properly. For now, it'd be good to do something that we can just adjust in visual code.

**Comment by GeorgeBellTMH** [comment]
I'd just add that whatever we do needs to work on the app under two situations 

1) user has selected a site - should show on homepage the next meeting time (or two/three of them potentially)

2) user has no site selected - there should likely be a component that shows which sites are meeting "soon" ie in the next week or so.

**Comment by itbel** [comment]
This was addressed in #1174 ...
